Objects/codeobject.c

```diff
@@ -241,8 +278,9 @@ intern_constants(PyObject *tuple, int *modified)
 
         // Intern non-string constants in the free-threaded build
         _PyThreadStateImpl *tstate = (_PyThreadStateImpl *)_PyThreadState_GET();
-        if (!_Py_IsImmortal(v) && !PyCode_Check(v) &&
-            !PyUnicode_CheckExact(v) && !tstate->suppress_co_const_immortalization)
+        if (!_Py_IsImmortal(v) && !PyUnicode_CheckExact(v) &&
```

Comment:
Is the explicit check for unicode necessary, given that should_immortalize_constant also checks that? (For performance, perhaps?)

Reply:
The checks were in both places because I wrote `should_immortalize_constant` recursively for tuples, sets, and slices.

I've changed `should_immortalize_constant` so that it's no longer recursive and instead interns and immortalizes tuples, set, and slices if we've already immortalized their elements. That should still end up interning and immortalizing the same objects, but avoids the seemingly redundant checks.